packages/wallet/src/services/WalletAssetProvider.ts

```diff
@@ -0,0 +1,214 @@
+import { Asset, AssetProvider, Cardano, GetAssetArgs, GetAssetsArgs, HealthCheckResponse } from '@cardano-sdk/core';
+import { Assets } from '../types';
+import { Logger } from 'ts-log';
+import { Observable, firstValueFrom } from 'rxjs';
+import { isNotNil } from '@cardano-sdk/util';
+
+export interface AssetProviderContext {
+  assetProvider: AssetProvider;
+  assetInfo$: Observable<Assets>;
+  tx: Cardano.Tx;
+  logger: Logger;
+}
+
+const tryCip68Metadata = (
+  policyId: Cardano.PolicyId,
+  name: Cardano.AssetName,
+  tx: Cardano.Tx,
+  logger: Logger
+): Asset.NftMetadata | null => {
+  const decoded = Asset.AssetNameLabel.decode(name);
+  if (decoded?.label === Asset.AssetNameLabelNum.UserNFT) {
+    const referenceAssetId = Cardano.AssetId.fromParts(
+      policyId,
+      Asset.AssetNameLabel.encode(decoded.content, Asset.AssetNameLabelNum.ReferenceNFT)
+    );
+
+    // TODO: It is possible that the reference NFT is not in one of the outputs of the transaction and was previously minted. We
+    // need a way to find the reference NFT TxOut from the current active UTXO set on the network.
+    for (const output of tx.body.outputs) {
+      if (output.value.assets?.get(referenceAssetId)) {
+        return Asset.NftMetadata.fromPlutusData(output.datum, logger);
+      }
+    }
+  }
+
+  return null;
+};
+
+const getNftMetadata = (
+  name: Cardano.AssetName,
+  policyId: Cardano.PolicyId,
+  tx: Cardano.Tx,
+  logger: Logger
+): Asset.NftMetadata | null => {
+  // First, try CIP-25
+  let metadata = tx.auxiliaryData?.blob
+    ? Asset.NftMetadata.fromMetadatum({ name, policyId }, tx.auxiliaryData.blob, logger)
+    : undefined;
+
+  // If metadata is not found, try CIP-68
+  if (!metadata) {
+    metadata = tryCip68Metadata(policyId, name, tx, logger);
+  }
+
+  return metadata;
+};
+
+const createAssetInfo = (assetId: Cardano.AssetId, amount: bigint, tx: Cardano.Tx, logger: Logger): Asset.AssetInfo => {
+  const name = Cardano.AssetId.getAssetName(assetId);
+  const policyId = Cardano.AssetId.getPolicyId(assetId);
+  const assetInfo: Asset.AssetInfo = {
+    assetId,
+    fingerprint: Cardano.AssetFingerprint.fromParts(policyId, name),
+    name,
+    policyId,
+    quantity: 0n,
+    supply: amount
```

Comment:
`quantity` was renamed to `supply` and is the same value, but is a deprecated field
```suggestion
    quantity: amount,
    supply: amount
```

Reply:
done here https://github.com/input-output-hk/cardano-js-sdk/pull/1075/commits/c39bcf81694122eb4d49fef57ab00e41b3480634